tests/nodeos_read_terminate_at_block_test.py

```diff
@@ -102,12 +97,62 @@ def executeTest(cluster, producerNode, testNodeId, testNodeArgs, resultMsgs):
         resultMsgs.append(resultDesc)
 
         # Kill node after use.
-        if not testNode.killed:
+        if testNode and not testNode.killed:
             assert testNode.kill(signal.SIGTERM)
 
     return testResult
 
 
+def checkStatus(testNode, testNodeArgs):
+    """Test --terminate-at-block stops at the correct block."""
+    Print(" ".join([
+        "The test node has begun receiving from the producing node and",
+        "is expected to stop at or little bigger than the block number",
+        "specified here: ",
+        testNodeArgs
+    ]))
+
+    # Read block information from the test node as it runs.
+    head, lib = getBlockNumInfo(testNode)
+
+    Print("Test node head = {}, lib = {}.".format(head, lib))
+
+    if "irreversible" in testNodeArgs:
+        checkIrreversible(head, lib)
+    else:
+        checkHeadOrSpeculative(head, lib)
+
+
+def checkReplay(testNode, testNodeArgs):
+    """Test --terminate-at-block with --replay-blockchain."""
+    # node.getInfo() doesn't work when replaying the blockchain so a
+    # relaunch  combined with --terminate-at-block will appear to fail.
+    # In reality, the relaunch works fine and it will (hopefully)
+    # run until completion normally. Code below ensures it does.
+    Print(" ".join([
+        "Relaunch the node in replay mode. The replay should stop",
+        "at or little bigger than the block number specified here: ",
+        testNodeArgs
+    ]))
+
+    testNode.relaunch(chainArg="--replay-blockchain", cachePopen=True)
+
+    # Check node exited without error.
+    assert testNode.popenProc.returncode == 0
```

Comment:
I think you can rely only on `testNode.relaunch` return value. No need additional check

Reply:
Removed the return value check.